### PR TITLE
Add Automations tile to Flow page

### PR DESCRIPTION
**Pull request by whimsicallyson**
PR Checklist:

- [x] add a short description of what's changed to the top of the `CHANGELOG.md`
- [ ] add/update tests (or don't, for reasons explained below)

## Describe this PR
Adds an Automations tab to the flow page subnav, which shows all automations registered to that flow. Solves #864

### CODE REVIEW

File: src/pages/Flow/Automations-Tile.vue [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+<template>
+  <v-card class="pa-2 mt-2" tile>
+    <CardTitle title="Automations" icon="fa-random" />
+
+    <v-card-text class="full-height position-relative">
+      <v-row>
+        <v-col cols="12" class="pb-0">
+          <div
+            >Displaying automations for this flow. To see all automations or
```
Comment (ThatGalNatalie) [inline src/pages/Flow/Automations-Tile.vue:82]: Should we display a message here when there are no automations or will that be too redundant?

File: src/pages/Flow/Automations-Tile.vue [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    hooks() {
+      this.sortHooks()
+    }
+  },
+  methods: {
+    async sortHooks() {
+      this.loadCards = true
+      const hooks = this.hooks ? [...this.hooks] : []
+      let filtered = hooks.filter(
+        hook => hook.event_tags.flow_group_id === this.flow.flow_group_id
```
Comment (zhen0) [inline src/pages/Flow/Automations-Tile.vue:33]: Flow group ids are returned as an array here.  This means you are missing all state change hooks.

File: src/pages/Flow/Automations-Tile.vue [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        (a, b) => new Date(b.created) - new Date(a.created)
+      )
+      this.sortedHooks = sorted
+      this.loadCards = false
+    }
+  },
+  apollo: {
+    hooks: {
+      query() {
+        return require('@/graphql/Automations/hooks.gql')
```
Comment (zhen0) [inline src/pages/Flow/Automations-Tile.vue:62]: We could/should narrow this query down by excluding any agent SLA hooks

File: src/pages/Flow/Automations-Tile.vue (hunk 1 above)
Comment (zhen0) [reply src/pages/Flow/Automations-Tile.vue:82]: Agreed. We should have a 'No flows associated with this hook' message.

File: src/pages/Flow/Flow.vue [hunk 4]
```diff
@@ -482,6 +502,11 @@ export default {
         <v-icon>pi-schematic</v-icon>
       </v-btn>
 
+      <v-btn :input-value="tab == 'automations'" @click="tab = 'automations'">
+        Automations
+        <v-icon>fa-random</v-icon>
```
Comment (zhen0) [inline src/pages/Flow/Flow.vue:507]: This should be duo-tone as the dashboard one is.

File: src/pages/Flow/Automations-Tile.vue [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+      loadingKey: 'loadingHook',
+      update: data => data?.hook
+    }
+  }
+}
+</script>
+
+<template>
+  <v-card class="pa-2 mt-2" tile>
+    <CardTitle title="Automations" icon="fa-random" />
```
Comment (zhen0) [inline src/pages/Flow/Automations-Tile.vue:76]: This icon looks really big to me.  Could we try a smaller size and make it duo-tone.

**Review by zhen0 — CHANGES_REQUESTED** [review]
My main concern here is that the filter for state change hooks is broken.  Other smaller design comments.

File: src/pages/Flow/Automations-Tile.vue (hunk 5 above)
Comment (whimsicallyson) [reply src/pages/Flow/Automations-Tile.vue:76]: couldn't get it to work with the CardTitle component, so I made the title manually 😕

File: src/pages/Flow/Automations-Tile.vue [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+          >
+        </v-col>
+        <v-progress-circular
+          v-if="!sortedHooks.length && loadingHook > 0"
+          class="mx-auto my-4"
+          indeterminate
+          color="primary"
+        />
+        <v-col
+          v-if="!sortedHooks.length && loadingHook == 0"
```
Comment (zhen0) [inline src/pages/Flow/Automations-Tile.vue:108]: Should we also include !loadCards?

**Review by zhen0 — DISMISSED** [review]

File: src/pages/Flow/Automations-Tile.vue [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+      loadingKey: 'loadingHook',
+      update: data => data?.hook
+    }
+  }
+}
+</script>
+
+<template>
+  <v-card class="pa-2 mt-2" tile>
+    <v-list-item dense class="px-0" :to="params">
```
[the remark this answers is not in the document]
Comment (whimsicallyson) [reply src/pages/Flow/Automations-Tile.vue]: nothing, bad copy-paste :/

File: src/graphql/Automations/flow-only-hooks.gql [hunk 8]
```diff
@@ -0,0 +1,15 @@
+query Actions {
+  hook{
```
[the remark this answers is not in the document]
Comment (whimsicallyson) [reply src/graphql/Automations/flow-only-hooks.gql]: It was in my local file...wtf computer. Should be there for real now.

**Review by zhen0 — COMMENTED** [review]
Thanks for the changes @whimsicallyson LGTM!  Any comments before we merge @znicholasbrown ?

**Comment by znicholasbrown** [comment]
I haven't had a chance to look at this yet but will put up a review later today 👍🏻

**Comment by znicholasbrown** [comment]
@whimsicallyson could we disable the interactive styling with the component for the flow tab? Right now the automations hint that they can be clicked because of hover/focus styles but we don't allow that from that tab.

**Comment by whimsicallyson** [comment]
@znicholasbrown disabled! The change is on the card itself, but I think that's good because the styling was happening for uneditable automations on the main tab as well.

**Review by znicholasbrown — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**